incubator/hnc/api/v1alpha1/hnc_config.go

```diff
@@ -99,6 +109,35 @@ type HNCConfigurationList struct {
 	Items           []HNCConfiguration `json:"items"`
 }
 
+// HNCConfigurationCode is the machine-readable, enum-like type of `HNCConfigurationCondition.Code`.
+// See that field for more information.
+type HNCConfigurationCode string
+
+// HNCConfigurationCondition specifies the code and the description of an error condition.
+type HNCConfigurationCondition struct {
+	// Describes the condition in a machine-readable string value. The currently valid values are
+	// shown below, but new values may be added over time. This field is always present in a
+	// condition.
+	//
+	// All codes that begin with the prefix `crit` indicate that reconciliation has
+	// been paused for this configuration. Future changes of the configuration will be
+	// ignored by HNC until the condition has been resolved. Non-critical conditions
+	// typically indicate some kinds of error that HNC itself can ignore. However,
+	// the behaviors of some types might be out-of-sync with the users' expectations.
+	//
+	// Currently, the supported values are:
+	//
+	// - "critSingletonNameInvalid": the specified singleton name is invalid.
```

Comment:
Maybe add some additional guide information like users should either 1) rename it to "config" if this is the right and only existing `HNCConfiguration` resource they want to use or 2) delete it if the right "config" already exists?

Reply:
Good idea! I added another sentence to indicate what the correct name should be. It depends on the users if they want to delete the object that has the wrong name or not. When they rename it to "config" and do k apply -f  config.yaml  the "config" object will be updated with the contents in config.yaml.